### ISSUE TITLE
Add IsolationForest as an optional outlier detector

### ISSUE THREAD

**Issue by bijay-odyssey** [bijay-odyssey/edaprep#2]
`docs/design-rationale.md` §6 lists IsolationForest as an optional advanced outlier
method, but `preprocessing/outliers.py` only registers four detectors:

```python
_DETECTORS = {"iqr", "zscore", "modified_zscore", "percentile"}
```

All four are univariate. IsolationForest is multivariate — it can flag a row whose
individual values are each unremarkable but whose *combination* is not, which the
existing fences structurally cannot.

### What to do

Add `IsolationForestDetector` in `src/edaprep/preprocessing/outliers.py`.

The wrinkle: the `OutlierDetector` interface returns a `Bounds` (a per-column interval),
and a multivariate method has no such thing. Two honest options, and picking between
them is part of the task:

1. Keep it per-column (a degenerate use of IsolationForest, simple to fit the interface).
2. Add a parallel `MultivariateDetector` path that returns a row mask directly, and let
   `OutlierHandler` accept either.

Option 2 is more work and more correct. Say which you are doing in the PR.

- scikit-learn is an **optional** dependency: import it inside the method and raise a
  message naming `pip install "edaprep[advanced]"` if it is missing, the way
  `visualization/plots.py::_plt` does.
- Honour `random_state` from the context so results stay reproducible.

### Tests

- flags a point that is an outlier only in two dimensions jointly, which `IQRDetector`
  misses — that contrast is the reason the detector exists
- `pytest.importorskip("sklearn")`
- same `random_state` gives the same mask twice

---

**Before you start:** please comment to claim this issue, and check whether someone
already has. Two people independently fixed #14 eight hours apart because this note
was missing, and one of them had to be turned away. If an issue has been claimed but
looks stale, say so in the thread rather than opening a competing PR.

**Comment by vyakhya923-spec** [comment]
Hi! I'd like to work on this issue. I’ll first inspect the existing Outlier Detector/Outlier Handler architecture and then implement the Isolation Forest path with tests for the multivariate case and random state reproducibility.

**Comment by bijay-odyssey** [comment]
Assigned — thanks for picking it up.

Worth deciding the option-1-vs-2 fork before writing much code, since it's a bigger fork than it might look at first glance. `OutlierDetector.fit_bounds(values: np.ndarray) -> Bounds` (outliers.py) takes one column's values and returns one interval — every existing detector, and `OutlierHandler` itself, is built around "one `Bounds` per column, applied via `Bounds.mask()`." Option 1 (make IsolationForest fit that shape) means fitting it on a single column at a time, which is univariate IsolationForest — it would never see the joint structure across columns that's the entire reason to reach for it over IQR/z-score. That's not really "the interface, degenerately used," it's a different algorithm that happens to share a class name.

So option 2 is the right one, but it means `OutlierHandler` needs a second path that fits across the selected columns at once and produces a row mask directly, alongside (not replacing) the existing per-column `Bounds` path — `strategy` (report/clip/winsorize/impute/remove) still needs to mean something for a row-level flag, which is worth thinking through for each strategy value rather than assuming they all port over unchanged (does "clip" even make sense for a multivariate flag? worth saying what you land on in the PR).

No need to solve all of that before claiming — just flagging it so you're deciding it on purpose rather than discovering it three files in.

**Comment by sanjana658** [comment]
Hi! I'd like to work on this issue as my first open-source contribution. I plan to implement the multivariate detector path (option 2) and add the requested tests. Please let me know if that's okay.

**Comment by bijay-odyssey** [comment]
Welcome, and congratulations on the first open-source contribution — glad to have you.

This one's already claimed and assigned, though: @vyakhya923-spec picked it up on 19 Sep and is working the option-1-vs-2 question I laid out above. Nothing wrong with your plan (option 2 is in fact the right call, per that same comment) — just first here.

A few unclaimed issues that are good candidates for a first PR, roughly easiest first:

- **[#37](https://github.com/bijay-odyssey/edaprep/issues/37)** — `Plan.without_columns` silently drops steps it shouldn't. Self-contained: exact bug, exact fix shape, and the tests to write are all spelled out in the issue.
- **[#44](https://github.com/bijay-odyssey/edaprep/issues/44)** — timedelta columns get profiled inconsistently (two related bugs, same root cause, one issue).
- **[#38](https://github.com/bijay-odyssey/edaprep/issues/38)** — profiler quality checks scoped to a sample when they shouldn't be. A bit more judgment involved (there's a design choice to make, spelled out in the issue).
- **[#6](https://github.com/bijay-odyssey/edaprep/issues/6)** — export a fitted `Plan` as a scikit-learn `ColumnTransformer`. A real feature rather than a bug fix, if that's more your interest.

Comment on whichever one you want and I'll assign it — same as here, first to claim gets it, so if two of these interest you, say so and I'll point you to whichever's more likely to stay free.
